transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringIoHandler.java

```diff
@@ -196,12 +202,21 @@ void submitAndRunNow(long udata) {
         }
         SubmissionQueue submissionQueue = ringBuffer.ioUringSubmissionQueue();
         CompletionQueue completionQueue = ringBuffer.ioUringCompletionQueue();
-        if (submissionQueue.submit() > 0) {
+        if (submitAndClear(submissionQueue) > 0) {
             completionBuffer.drain(completionQueue);
             completionBuffer.processOneNow(this::handle, udata);
         }
     }
 
+    private int submitAndClear(SubmissionQueue submissionQueue) {
+        int submitted = submissionQueue.submit();
+
+        // Clear the iovArray as we can re-use it now as things are considered stable after submission:
+        // See https://man7.org/linux/man-pages/man3/io_uring_prep_sendmsg.3.html
+        iovArray.clear();
```

Comment:
Should we check whether the current io_uring supports IORING_FEAT_SUBMIT_STABLE?
https://github.com/axboe/liburing/issues/27
linux 5.5 already supports this flag, and since Netty’s baseline Linux version is 5.9, it seems we don’t need to worry about this?

Reply:
@dreamlike-ocean we already do https://github.com/netty/netty/pull/14996/files#diff-6448077164bee8d0404c69357049f590a8b66a4ab8a3555e5ffee25df3b5465fR86